File: backend/src/main/java/org/globe42/web/charges/ChargeTypeDTO.java
package org.globe42.web.charges;

import java.math.BigDecimal;

import org.globe42.domain.ChargeType;

/**
 * DTO of a charge type
 * @see ChargeType
 * @author JB Nizet
 */
public final class ChargeTypeDTO {
    private final Long id;
    private final String name;
    private final ChargeCategoryDTO category;
    private final BigDecimal maxMonthlyAmount;

    public ChargeTypeDTO(ChargeType chargeType) {
        this.id = chargeType.getId();
        this.name = chargeType.getName();
        this.category = new ChargeCategoryDTO(chargeType.getCategory());
        this.maxMonthlyAmount = chargeType.getMaxMonthlyAmount();
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public ChargeCategoryDTO getCategory() {
        return category;
    }

    public BigDecimal getMaxMonthlyAmount() {
        return maxMonthlyAmount;
    }
}

File: backend/src/main/kotlin/org/globe42/web/charges/ChargeTypeDTO.kt
package org.globe42.web.charges

import org.globe42.domain.ChargeType
import java.math.BigDecimal

/**
 * DTO of a charge type
 * @see ChargeType
 *
 * @author JB Nizet
 */
data class ChargeTypeDTO(
        val id: Long,
        val name: String,
        val category: ChargeCategoryDTO,
        val maxMonthlyAmount: BigDecimal?) {

    constructor(chargeType: ChargeType) : this(chargeType.id!!,
                                               chargeType.name!!,
                                               ChargeCategoryDTO(chargeType.category!!),
                                               chargeType.maxMonthlyAmount)
}
